src/components/ThirdPartyCollectionDetailPage/ThirdPartyCollectionDetailPage.tsx

```diff
@@ -201,15 +207,19 @@ export default class ThirdPartyCollectionDetailPage extends React.PureComponent<
                 </Column>
                 <Column align="right" shrink={false}>
                   <Row className="actions">
-                    <Button secondary compact className={`${areSlotsEmpty ? 'empty' : ''} slots`} onClick={this.handleBuySlot}>
-                      {t('third_party_collection_detail_page.slots', { amount: slots.toNumber() })}
-                      {areSlotsEmpty ? (
-                        <span className="buy-slots link" onClick={this.handleBuySlot}>
-                          {t('global.buy')}
-                        </span>
-                      ) : null}
+                    <Button
+                      loading={isLoadingAvailableSlots}
+                      secondary
+                      compact
+                      className={classNames({ empty: areSlotsEmpty && !isLoadingAvailableSlots, slots: !isLoadingAvailableSlots })}
+                      onClick={this.handleBuySlot}
+                    >
+                      <>
+                        {t('third_party_collection_detail_page.slots', { amount: thirdParty?.availableSlots })}
+                        {areSlotsEmpty ? <span className="buy-slots link">{t('global.buy')}</span> : null}
+                      </>
                     </Button>
-                    <CollectionPublishButton collection={collection} items={selectedItems} slots={slots.toNumber()} />
+                    <CollectionPublishButton collection={collection} items={selectedItems} slots={thirdParty?.availableSlots || 0} />
```

Comment:
is the third party always going to be defined in this component? couldn't this blow up otherwise?
Same goes for [PublishThirdPartyCollectionModal.tsx](https://github.com/decentraland/builder/pull/1782/files#diff-b480305804f9a96c5cd57aadd170b311245effbfa853a5ff6b0233a26a5fc2d2)

Reply:
That's a good question, it's not that clear in this component if it's always going to be defined. There could be a race condition where the `collection` has been fetch but the `thirdParties` request has not been resolved and thus the `thirdParty` might still be `null`.

I made the following changes to address that:

In the `render` method, before calling `this.renderPage`, I've added the check if `thirdParty` is defined and I'm passing that down to the method instead of reading it from props (to avoid it being typed as `| null`)
```
{shouldRender && thirdParty ? this.renderPage(thirdParty) : <NotFound />}
```
and also moved the `getCollectionType(collection) === CollectionType.THIRD_PARTY` to the `.contatiner` file since I think that having a `thirdParty` object defined (in the render method) covers that case and it would be better to not call the `getCollectionThirdParty` selector (in the container) at all if the collection is not a TP one.

------


For [PublishThirdPartyCollectionModal.tsx](https://github.com/decentraland/builder/pull/1782/files#diff-b480305804f9a96c5cd57aadd170b311245effbfa853a5ff6b0233a26a5fc2d2) I think we're OK because in order to get that modal opened, we need to render the detail page first (and so `thirdParty` would be defined).

Reply:
Perfect on both fronts! thanks for the detailed explanation